From 7dbf8d353aae0259f924c67c4b1dd78079ce2f5d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9bastien=20Wilmet?= <swilmet@mailfence.com>
Date: Fri, 12 Apr 2024 18:25:51 +0200
Subject: [PATCH] Disable translations

When printing messages on the terminal, UTF-8 encoded characters are
still correctly displayed. Because "C.UTF-8" is set, not just "C".
---
 gedit/gedit.c | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/gedit/gedit.c b/gedit/gedit.c
index f5aba8c1e..d12efe677 100644
--- a/gedit/gedit.c
+++ b/gedit/gedit.c
@@ -101,7 +101,17 @@ gedit_w32_unload_private_dll (void)
 #endif /* G_OS_WIN32 */
 
 static void
-setup_i18n (void)
+setup_i18n_first_part (void)
+{
+	/* Disable translations because some underlying modules are not on
+	 * l10n.gnome.org or equivalent.
+	 * See the docs of g_setenv(), needs to be called very early in main().
+	 */
+	g_setenv ("LC_ALL", "C.UTF-8", TRUE);
+}
+
+static void
+setup_i18n_second_part (void)
 {
 	const gchar *dir;
 
@@ -144,6 +154,8 @@ main (int argc, char *argv[])
 	GeditApp *app;
 	gint status;
 
+	setup_i18n_first_part ();
+
 #if OS_MACOS
 	type = GEDIT_TYPE_APP_OSX;
 #elif defined G_OS_WIN32
@@ -162,7 +174,7 @@ main (int argc, char *argv[])
 	 */
 	gedit_dirs_init ();
 
-	setup_i18n ();
+	setup_i18n_second_part ();
 	setup_pango ();
 	tepl_init ();
 	factory = gedit_factory_new ();
-- 
GitLab

